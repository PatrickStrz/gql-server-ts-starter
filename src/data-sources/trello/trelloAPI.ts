import { RESTDataSource, RequestOptions } from "apollo-datasource-rest"

export class TrelloAPI extends RESTDataSource {
  public baseURL = "https://api.trello.com/1/"

  public willSendRequest(request: RequestOptions) {
    request.params.set("key", process.env.TRELLO_API_KEY)
    request.params.set("token", process.env.TRELLO_API_TOKEN)
  }
  public async boards(): Promise<any> {
    return this.get("/members/me/boards/")
  }
}
