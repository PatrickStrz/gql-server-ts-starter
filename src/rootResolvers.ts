export const resolvers = {
  Query: {
    /* tslint:disable-next-line */
    books: async (_0: any, _1: any, { dataSources }: any) => {
      console.log(await dataSources.trelloAPI.boards())
      return books
    },
    /* tslint:disable-next-line */
    boards: async (_0: any, _1: any, { dataSources }: any) => {
      return await dataSources.trelloAPI.boards()
    },
  },
}

const books = [
  {
    title: "Harry Potter and the Chamber of Secrets",
    author: "J.K. Rowling",
  },
  {
    title: "Jurassic Park",
    author: "Michael Crichton",
  },
]
