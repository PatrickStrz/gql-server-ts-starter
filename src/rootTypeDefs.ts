import { gql } from "apollo-server"

export const typeDefs = gql`
  type Book {
    title: String
    author: String
  }

  type Board {
    id: ID!
    name: String
  }

  type Query {
    books: [Book]
    boards: [Board]
  }
`
