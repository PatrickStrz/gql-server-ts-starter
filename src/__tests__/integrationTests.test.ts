import { createTestClient } from "apollo-server-testing"
import { ApolloServer } from "apollo-server"
import { TrelloAPI } from "../data-sources/trello/trelloAPI"
import { resolvers } from "../rootResolvers"
import { typeDefs } from "../rootTypeDefs"
import { gql } from "apollo-server"

const ALL_BOARDS_QUERY = gql`
  {
    boards {
      id
    }
  }
`

// @ts-ignore
TrelloAPI.prototype.get = jest.fn(() => [{ id: 1 }, { id: 5 }])

export const dataSources = () => {
  return {
    trelloAPI: new TrelloAPI(),
  }
}

it("fetches single launch", async () => {
  // create a test server to test against, using our production typeDefs,
  // resolvers, and dataSources.
  const server = new ApolloServer({
    typeDefs,
    resolvers,
    dataSources,
  })
  // use the test server to create a query function
  // @ts-ignore
  const { query } = createTestClient(server)
  // run query against the server and snapshot the output
  const res = await query({ query: ALL_BOARDS_QUERY })
  // @ts-ignore
  expect(res).toMatchSnapshot()
  const b = { resolvers, typeDefs }

  expect(b).toBeTruthy()
})
