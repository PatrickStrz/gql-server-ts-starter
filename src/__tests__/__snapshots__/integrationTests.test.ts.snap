// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches single launch 1`] = `
Object {
  "data": Object {
    "boards": Array [
      Object {
        "id": "1",
      },
      Object {
        "id": "5",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
